storage-bigtable/src/bigtable.rs

```diff
@@ -265,7 +274,8 @@ impl BigTableConnection {
     {
         retry(ExponentialBackoff::default(), || async {
             let mut client = self.client();
-            Ok(client.put_bincode_cells(table, cells).await?)
+            let result = client.put_bincode_cells(table, cells).await;
+            check_table_not_found(result)
```

Comment:
how about something like this
```suggestion
           result.map_err(to_backoff_err)
```